### PR TITLE
Attempt to fix js-delete

### PR DESCRIPTION
**Pull request by themmes**
Fix for https://github.com/digitallyinduced/ihp/issues/361, changes..

- add `handleClick` only to elements with `js-delete` to limit scope to `<a>` (instead of `document`)
- change `target` to `currentTarget`

This seems to work, however I get issues when trying to delete right after being routed from `New` or `Edit`. For some reason it requires a refresh of the page before working correctly. Maybe you can verify whether that is due to my project?

### CODE REVIEW

**Comment by mpscholten** [comment]
Thanks! Looks good so far :)

By using `document.querySelectorAll(".js-delete")` the listener on attachs to elements available at the first page load. When using a form or link, we're using turbolinks for navigation (it's part of the default IHP setup), therefore we have to reattach the listener after turbolinks applied changes to the DOM. We're doing this for all the other helpers [already here in line 11](https://github.com/digitallyinduced/ihp/blob/master/lib/IHP/static/helpers.js#L11). Due to use of morphdom we also need to make sure that we don't attach our event listener twice (the DOM node might still be the same because of the diff approach by morphdom). This might be tricky :)

File: lib/IHP/static/helpers.js [hunk 1]
```diff
@@ -42,20 +42,20 @@ function initTime() {
 }
 
 function initDelete() {
-    document.addEventListener('click', handleClick);
+    document.querySelectorAll(".js-delete").forEach(e => e.addEventListener('click', handleClick));
```
Comment (mpscholten) [inline lib/IHP/static/helpers.js:45]: Could we use boring JS here? :) This way these helpers support even older browsers. (Only in the JS files used by the dev tools, we can use more modern JS)

```suggestion
    document.querySelectorAll(".js-delete").forEach(function (e) { e.addEventListener('click', handleClick) });
```

Comment (themmes) [reply lib/IHP/static/helpers.js]: Makes sense!

**Comment by themmes** [comment]
That should do it!

**Comment by themmes** [comment]
There might still be something weird, if you cancel the confirmation prompt it asks you a second time. Any clue why that happens?

**Comment by mpscholten** [comment]
I guess the callback is registered multiple times on the link due to turbolinks events being triggered multiple times. You could add a `console.log` to the callback in  `document.querySelectorAll(".js-delete").forEach(function(e) {` to verify this :)

**Comment by themmes** [comment]
Have not been able to finish this, after my holidays (2 weeks) I'll pick this up!

**Comment by themmes** [comment]
@mpscholten I'm sorry, I have tried but I am not able to confirm or exclude it's caused by turbolinks.

**Comment by themmes** [comment]
@mpscholten I feel like handing in my programmers-license, but I really can't lay my finger on this issue. It's black magic to me 🧙 !

**Comment by mpscholten** [comment]
:) no worries, will take a look on monday and maybe can provide a fix 👍

**Pull request merged**